pkg/agent/proxy/proxier.go

```diff
@@ -403,14 +414,18 @@ func (p *proxier) installServices() {
 			externalPolicyLocal = true
 		}
 
-		clusterEndpoints, localEndpoints, allReachableEndpoints := p.categorizeEndpoints(endpointsToInstall, svcInfo)
+		var mcsLocalService *openflowtypes.ServiceGroupInfo
+		clusterEndpoints, localEndpoints, allReachableEndpoints, mcsLocalService := p.categorizeEndpoints(endpointsToInstall, svcInfo, svcPortName, serviceCIDRIPv4)
 		// If there are new Endpoints, Endpoints installed should be updated.
 		for _, endpoint := range allReachableEndpoints {
 			if _, ok := endpointsInstalled[endpoint.String()]; !ok { // There is an expected Endpoint which is not installed.
 				needUpdateEndpoints = true
 				break
 			}
 		}
+		if mcsLocalService != nil {
+			needUpdateEndpoints = true
+		}
```

Comment:
This means the whole endpoints of this service will be updated as long as there is any service/endpoints change in the whole cluster and when the peridiocal sync happens (30s).

Why it must update endpoints when there is a local service endpoint?